### ISSUE TITLE
Sqlite bigIncrements generating integer auto increment

### ISSUE THREAD

**Issue by mahendraHegde** [directus/directus#20889]
### Describe the Bug
Choosing autoincrement bigInt as id actually creates integer auto increment.

### To Reproduce

1. select sqlite driver
2. create a new collection with id as bigInt with auto increment
3. go back to field definition in UI and see, it will be reset to integer

### Directus Version

10.13.3

### Hosting Strategy

Self-Hosted (Docker Image)

**Comment by mahendraHegde** [comment]
i can fix the issue, lmk what solution do you prefer.

**Comment by br41nslug** [comment]
I dont think this has anything to do with SQLite specifically but rather just BigInt handling in filters in general.

I am unfortunately not able to reproduce the described resetting of the stored type in SQLite. Are there extra steps required for this reset to happen? I've tried navigating and restarting the instance

https://github.com/directus/directus/assets/9389634/bfbe5924-20aa-46f4-9604-b9d1202c8e37

Closing this for now until we can get this reproduceable.

**Comment by mahendraHegde** [comment]
https://screenapp.io/app/#/shared/d98c829a-7383-4b0e-ae44-c84f4a021641 
@br41nslug  it seems it only happening for bigInt autoincrement
initially i thought its an issue with knex, as query generated was ```create table `test2` (`id` integer not null primary key autoincrement)'```

but when i tried directly ```create table `test2` (`id` bigint not null primary key autoincrement)'```  in sql console sqlite rejects it with error `AUTOINCREMENT is only allowed on an INTEGER PRIMARY KEY`

**Comment by br41nslug** [comment]
You're very right explicitly when using it as primary key i was able to reproduce it 😄 Sounds like SQLite simply does not support it tho 🤔 Perhaps throwing an error there when saving the collection is a better approach than failing gracefully to `integer`

**Comment by mahendraHegde** [comment]
also may be 
-  disabling/hiding bigInt increment option in ui for sqlite

**Comment by br41nslug** [comment]
> disabling/hiding bigInt increment option in ui for sqlite

I don't think we should go that route, the App itself is not aware of what backend database was used and shouldn't be i think

**Comment by phazonoverload** [comment]
The right place for this note in docs will be the yet-to-be created #20610 Should we consider this a docs issue @br41nslug?

**Comment by br41nslug** [comment]
Yeah i think educating about this is the best way to go here since this is such an edge case and it is still working as intended because SQLite makes no distinction between `int` or `bigInt`.
